app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ArticlesController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", except: [:index, :show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article = Article.find(params[:id])
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    # @article = Article.new(title: "Amazing Animals", body: "Show of top 10 Animals"title: "Amazing Animals", body: "Show of top 10 Animals")
+    @article = Article.new(article_params)
+    if @article.save
```

Comment:
What happens if we use `@article.save!` ?

Reply:
Reading documentation about `ActiveRecord` and `save` method, with the bang expressión, executes validations and if a value is missing or wrong raise an exception.

Reply:
What should you do if it raises an exception?

Reply:
In this case, i changed from a conditional to a rescue statement, and works fine!
```
begin
      @article.save!
rescue
      render :new
else
      redirect_to(article_path(@article))
 end
```